### ISSUE TITLE
Does not properly mock interface with extension functions

### ISSUE THREAD

**Issue by carltonwhitehead** [mockative/mockative#30]
Example:

```kotlin
interface Foo {
  fun Widget.bar()
}

@Mock val foo = mock(classOf<Foo>())
```

The generated mock class contains an override for the extension function without the receiver class declaration (`Widget.`)

This produces errors along the lines of:
```no-highlight
e: Class 'FooMock' is not abstract and does not implement abstract member public abstract fun Widget.bar(): Unit defined in Foo
e: 'bar' overrides nothing
```

I'm seeing this with mockative version 1.0.7. I haven't tested with the latest version of mockative because our application isn't upgraded to Kotlin 1.6 yet. I looked through the changes and issue tracker and didn't see anything suggesting this was a known issue.

**Comment by Nillerr** [comment]
Hi @carltonwhitehead 

Thanks for reporting this issue, and apologies for not getting back to you sooner.

I had not considered functions on interfaces with extension receivers in the original generator. I have just released version `1.2.0` of Mockative, which features a complete rewrite of the code generation using [KotlinPoet](https://square.github.io/kotlinpoet/), in which I have included code to catch this specific case. If you have already upgraded to Kotlin 1.6.0, please try this new version and reopen this issue if the problem remains 🙂

**Comment by Nillerr** [comment]
Hi again

I discovered an issue in `1.2.0` causing the generated code to be invalid. I have resolved this issue in `1.2.1`, which is on its way out.
